pkg/trace/writer/trace.go

```diff
@@ -213,29 +213,12 @@ func (w *TraceWriter) flush() {
 			log.Errorf("gzip.NewWriterLevel: %d", err)
 			return
 		}
-		gzipw.Write(b)
+		if _, err := gzipw.Write(b); err != nil {
+			log.Errorf("Error gzipping trace payload: %v", err)
+		}
 		gzipw.Close()
```

Comment:
Does close return an error too? I feel like it usually does for compressors

Reply:
I guess it does. I added a log for it, but didn't make it prevent anything in case of error.